src/main/java/org/spongepowered/api/command/parameter/managed/ValueCompleter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command.parameter.managed;
+
+import org.spongepowered.api.command.parameter.ArgumentReader;
+import org.spongepowered.api.command.parameter.CommandContext;
+
+import java.util.List;
+
+/**
+ * Defines the completions for a parameter.
+ */
+@FunctionalInterface
+public interface ValueCompleter {
+
+    /**
+     * Gets valid completions for this command.
+     *
+     * @param reader The {@link ArgumentReader.Immutable} containing the arguments
+     *               that needs to be completed.
+     * @param context The {@link CommandContext} that contains the parsed arguments
+     */
+    List<String> complete(ArgumentReader.Immutable reader, CommandContext context);
```

Comment:
reader -> arguments

Reply:
1. No description of the return value is specified.
2. Maybe replace `List` to `Set`? The proposed values ​​must be unique.

Reply:
Must be a `List`, because the order matters.

Reply:
There is of course `SortedSet`, but it will probably be inconvenient. :smile: